src/Workspaces/Remote/ServiceHub/Host/ChecksumSynchronizer.cs

```diff
@@ -47,14 +51,17 @@ public async ValueTask SynchronizeSolutionAssetsAsync(Checksum solutionChecksum,
 
                     solutionChecksumObject.AddAllTo(checksums);
                     checksums.Remove(solutionChecksumObject.Checksum);
-                    await _assetProvider.SynchronizeAssetsAsync(assetHint: AssetHint.None, checksums, cancellationToken).ConfigureAwait(false);
+                    await _assetProvider.SynchronizeAssetsAsync(assetHint: AssetHint.None, checksums, results: null, cancellationToken).ConfigureAwait(false);
```

Comment:
so we could pass a dictionary here that we then read in below.  but i didn't feel like it was worth it.  the data will normally always stay in the asset provider.  and the cahnge below to use GetAssetAsync will normally find it, or *rarely* then go and reretrieve it in the rare unlucky case.  that saves a potentially large dictionary alloc here (even if pooled).